wordpress-mobile/WordPress-iOS
# Issue/5552 Helpshift deep linking

Fixes #5552. Adds the ability to deep link to Helpshift FAQs. This should be the first step to providing the user with contextual help (for example, presenting login FAQs from the login screen). 

To implement this, I took a bunch of code out of `SupportViewController` into a more useful `HelpshiftPresenter` class, which can be used to present Helpshift windows from other places. 

I also replaced our custom `HelpshiftConfig.plist` (for Helpshift appearance) with the template one from the Helpshift SDK and used it to only change the status bar appearance. Our existing one was named incorrectly (it should be `HelpshiftCustomConfig.plist`, so it was never actually being used. I took a quick look at applying it, but I think things look better without those styles so I ditched it.

To test:

Run the app, then head to Safari and enter the URL for an FAQ. Note that it uses the `WPCOM_SCHEME` build setting for the URL scheme – so in the case of a debug build, it'll be `wpdebug`. Here are some example URLs:

`wpdebug://faq/47`
`wpdebug://faq/58`
`wpdebug://faq/114`

The number is the 'publish ID' taken from Helpshift.

Check that the FAQ window opens in various situations – signed out, signed in, etc.

Needs review: @koke
